src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -3296,4 +3309,86 @@ private function prepare_taxonomy_limit_schema( array $query_params ) {
 
 		return $query_params;
 	}
+
+	/**
+	 * Retrieves post counts for the post type.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_count() {
+		$counts = wp_count_posts( $this->post_type );
+		$data   = array();
+
+		if ( ! empty( $counts ) ) {
+			/*
+			 * The fields comprise all non-internal post statuses,
+			 * including any custom statuses that may be registered.
+			 * 'trash' is an exception, so if it exists, it is added separately.
+			 */
+			$post_stati = get_post_stati( array( 'internal' => false ) );
+
+			if ( get_post_status_object( 'trash' ) ) {
+				$post_stati[] = 'trash';
+			}
+			// Include all public statuses in the response if there is a count.
+			foreach ( $post_stati as $status ) {
+				if ( isset( $counts->$status ) ) {
```

Comment:
The above suggestion had an extra parenthesis:

`if ( rest_is_field_included( $status, $fields ) ) && isset( $counts->$status ) ) {`

As I understand it, this will check against `$this->get_item_schema()` so the fields won't exist.

`/counts` requires a unique schema.

I'm finding the problem is that the `schema` callback when registering a route is never fired, unless it's an [OPTIONS request](https://github.com/WordPress/wordpress-develop/blob/6106332617128820c0ee5132c34ff1cf6f785d1a/src/wp-includes/rest-api.php#L825) ([the context is 'help](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/class-wp-rest-server.php#L1627)'). So I'm not sure the REST API is set up to allow a response to have two separate schemas.

I guess this is a good thing. 

It's part of the reason why I chose to add a separate endpoint in the first place https://github.com/WordPress/gutenberg/pull/66294

There is the option to add a `counts` field to the existing schema. 🤔

Reply:
> There is the option to add a counts field to the existing schema

Integrating it into the current item schema isn't appropriate as it's not an item, it's related to the collection